src/smc-webapp/printing.coffee

```diff
@@ -577,7 +617,7 @@ class SagewsPrinter extends Printer
                     console.debug("write_text_file_to_project.resp: '#{resp}'")
                     cb?(err)
 
-        async.parallel([sagews_data, process_lines], finalize)
+        async.parallel([sagews_data, process_lines, embed_videos], finalize)
 
```

Comment:
Stupid annoying overall question: why is this all written using jquery and direct construction of HTML in the DOM?  Couldn't it be done in a cleaner, more easily debugable and maintained way by using React instead?   I know React is harder to write, but I think it is easier to debug, improve, and reason about, especially for non-authors...  Is there a good reason for not using react?